File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/subscriber/SubscriptionZoneMessage.scala





















// Generated on 05/03/2013 10:35:00
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait SubscriptionZoneMessage extends MessageDeserializer {
  val messageId = 5573

  protected trait Message extends BaseMessage with Serializable {
    val messageId = SubscriptionZoneMessage.messageId

    val active: Boolean

    def serialize(buf: Buffer) {
      buf.writeBoolean(active)
    }

  }

  protected def create(active: Boolean): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val active = buf.readBoolean()

    create(active)
  }

}

object SubscriptionZoneMessage extends SubscriptionZoneMessage {
  case class MessageImpl(active: Boolean) extends Message
  type Target = MessageImpl

  override def create(active: Boolean) = MessageImpl(active)

  def apply(active: Boolean) = create(active)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/SubscriptionZoneMessage.java


// Generated on 05/08/2013 19:37:58
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class SubscriptionZoneMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5573;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public boolean active;
    
    public SubscriptionZoneMessage() { }
    
    public SubscriptionZoneMessage(boolean active) {
        this.active = active;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeBoolean(active);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        active = buf.readBoolean();
    }
    
}
